pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCuda.hpp

```diff
@@ -499,6 +499,16 @@ class TNCuda : public TNCudaBase<PrecisionT, Derived> {
                       projected_mode_values, numHyperSamples);
     }
 
+    /**
+     * @brief Get a const vector reference of sitesExtents_.
+     *
+     * @return const std::vector<std::vector<std::size_t>>
+     */
+    [[nodiscard]] auto getSitesExtents()
+        -> const std::vector<std::vector<std::size_t>> & {
+        return sitesExtents_;
+    }
+
   protected:
```

Comment:
Could you merge protected and private sections to avoid the class scoping fragmentation?

Reply:
Unfortunately, it is not possible to merge protected and private because `MPSTNCuda` and `ExactTNCuda` are inherited from TNCuda class  :confused:

Reply:
No, I meant can we have one single `private` and one `protected` instead of multiple in this class?

Reply:
Done!